frontend/src/__tests__/unittest/signup/Signup.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { waitFor } from '@testing-library/react';
+import { vi } from 'vitest';
+import userEvent from '@testing-library/user-event';
+import { formSetup } from './Client-util';
+import {
+  INVALID_CONFIRMPWD_ERRORMESSAGE,
+  INVALID_EMAIL_ERRORMESSAGE,
+  INVALID_PWD_ERRORMESSAGE,
+  INVALID_USERNAME_ERRORMESSAGE,
+} from '@/constants/Stringconstant';
+const mockedUseNavigate = vi.fn();
+vi.mock('react-router-dom', async () => {
+  const mod = await vi.importActual<typeof import('react-router-dom')>('react-router-dom');
```

Comment:
Didn't get the use of it, can you kindly explain?

Reply:
@krishnaacharyaa  this is to mock the useNavigate hook from React Router DOM . On the successful signup the user is redirected to the home page. to test this case we have used this                                                                                                    const mockedUseNavigate = vi.fn();                                                                                                                                              This will create a mock function                                                                                                                                                      vi.mock('react-router-dom', async () => {
  const mod = await vi.importActual<typeof import('react-router-dom')>('react-router-dom');
  return {
    ...mod,
    useNavigate: () => mockedUseNavigate,
  };
});                                                                                                                                                                                                   This will override the real usenavigation function with our mocked function . so when we execute the test whenever the usenavigation function is called the mocked function gets called and we can find out this by                                                     expect(mockedUseNavigate).toHaveBeenCalledTimes(1);